### DISCUSSION TITLE
[Feature Request] Comic Support (zip files and divina profile via OPDS v2)

### DISCUSSION THREAD

**Discussion by beville** [edrlab/thorium-reader#2636] [Developer corner]
It would be awesome if this application also supported comics.  

The basic "cbz" file is just zip archive of images with an implicit order based on image file names.  I think the OPDS 2.0 acquisition link is of type "application/vnd.comicbook+zip"

Also, there is an OPDS 2 webpub profile called "divina" that supports a page-by-page stream of publications, that is documented here:  https://github.com/readium/webpub-manifest/blob/master/profiles/divina.md .  There are a number of self-hosted OPDS servers that are starting to add support for this feature, including Komga (komga.org)

**Comment by danielweck** [comment]
CBZ is supported at the parser level but there is currently no image-only rendering engine in Thorium.
The Divina file format is supported as a zipped publication package, and Thorium uses a third-party dedicated rendering engine for it.
Although some underlying component in the software stack supports resource streaming, Thorium itself was designed for packaged publications that are imported into a reserved filesystem and streamed from a local server. This part of the software architecture would need to be refactored in order to support resource streaming from remote servers.
PS: regarding CBZ comics, perhaps we could convert the images and metadata format into Divina and just rely on the existing Divina rendering engine?
